sdk/clientcore/core/src/main/java/io/clientcore/core/credentials/oauth/OAuthTokenProvider.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.credentials.oauth;
+
+import io.clientcore.core.credentials.AccessToken;
+
+/**
+ * <p>
+ * Token Credential interface serves as a fundamental component for managing and providing access tokens.
+ * </p>
+ *
+ * <p>
+ * The {@link OAuthTokenProvider} interface, offers API to retrieve an access token that can be used for
+ * bearer token authentication. The scopes parameter specified as part of {@link OAuthTokenProvider} represents
+ * the resources or permissions required for the token.
+ * </p>
+ *
+ * <p>
+ * By utilizing the OAuth Token Provider interface, you can abstract the authentication logic away from your
+ * application code. This allows for flexibility in choosing authentication mechanisms and simplifies the management
+ * of access tokens, including token caching and refreshing.
+ * </p>
```

Comment:
I might just delete this. If anything I think this should go into a conceptual section in the package-info for `credentials`.